modules/reference/pages/cluster-properties.adoc

```diff
@@ -1,1796 +1,5372 @@
+
+=== raft_heartbeat_disconnect_failures
+
+After how many failed heartbeats to forcibly close an unresponsive TCP connection.  Set to 0 to disable force disconnection.
+
+*Requires Restart:* No
+
+*Nullable:* No
+
+*Visibility:* `tunable`
+
+*Type:* integer
+
+*Default:* `3`
+
+---
+
+=== raft_heartbeat_interval_ms
+
+Milliseconds for raft leader heartbeats.
+
+*Requires Restart:* No
+
+*Nullable:* No
+
+*Visibility:* `tunable`
+
+*Type:* integer
+
+*Accepted values:* [`-17592186044416`, `17592186044415`]
+
+*Default:* `std::chrono::milliseconds(150)`
+
+---
+
+=== raft_heartbeat_timeout_ms
+
+Raft heartbeat RPC timeout.
+
+*Requires Restart:* No
+
+*Nullable:* No
+
+*Visibility:* `tunable`
+
+*Type:* integer
+
+*Accepted values:* [`-17592186044416`, `17592186044415`]
+
+*Default:* `3s`
+
+---
+
+=== raft_io_timeout_ms
+
+Raft I/O timeout.
+
+*Requires Restart:* No
+
+*Nullable:* No
+
+*Visibility:* `tunable`
+
+*Type:* integer
+
+*Accepted values:* [`-17592186044416`, `17592186044415`]
+
+*Default:* `10000ms`
+
+---
+
+=== raft_learner_recovery_rate
+
+Raft learner recovery rate limit. Throttles the rate of data communicated to nodes (learners) that need to catch up to leaders.
+
+This rate limit is placed on a node sending data to a recovering node. Each sending node will be limited to this rate. The recovering node will accept data as fast as possible according to the combined limits of all healthy nodes in the cluster. For example, if two nodes are sending data to the recovering node, and `raft_learner_recovery_rate` is 100 MB/sec, then the recovering node will recover at a rate of 200 MB/sec.
+
+*Units*: bytes per second.
+
+*Requires Restart:* No
+
+*Nullable:* No
+
+*Visibility:* `tunable`
+
+*Type:* integer
+
+*Default:* `104857600`
+
+---
+
+=== raft_max_concurrent_append_requests_per_follower
+
+Maximum number of concurrent append entries requests sent by leader to one follower.
+
+*Requires Restart:* No
+
+*Nullable:* No
+
+*Visibility:* `tunable`
+
+*Type:* integer
+
+*Accepted values:* [`0`, `4294967295`]
+
+*Default:* `16`
+
+---
+
+=== raft_max_recovery_memory
+
+Max memory that can be used for reads in raft recovery process by default 15% of total memory.
+
+*Requires Restart:* No
+
+*Nullable:* Yes
+
+*Visibility:* `tunable`
+
+*Type:* integer
+
+*Default:* `null`
+
+---
+
+=== raft_recovery_concurrency_per_shard
+
+How many partitions may simultaneously recover data to a particular shard. This is limited to avoid overwhelming nodes when they come back online after an outage.
+
+*Requires Restart:* No
+
+*Nullable:* No
+
+*Visibility:* `tunable`
+
+*Type:* integer
+
+*Default:* `64`
+
+---
+
+=== raft_recovery_default_read_size
+
+Default size of read issued during raft follower recovery.
+
+*Requires Restart:* No
+
+*Nullable:* No
+
+*Visibility:* `tunable`
+
+*Type:* integer
+
+*Default:* `524288`
+
+---
+
+=== raft_recovery_throttle_disable_dynamic_mode
+
+Disables dynamic rate allocation in recovery throttle (advanced).
```

Comment:
**ENG:** This property is not mentioned anywhere in the RP docs. The cryptic description here suggests that it is an advanced property, but does not explain what that means, when it should be used, etc. In what context would a user specify this property, and why? Also, "...in recovery throttle" is unclear--can we get clearer wording on this? Should it be "...during recovery throttle"?